### PR TITLE
async blob exchange, dht, stream, and blob rewrite/refactor

### PR DESCRIPTION
**Pull request by jackrobison**
Removes the following requirements

    twisted
    treq
    requests
    txJSON-RPC

Removes `twisted.trial` based testing, replaced with `unittest`

### CODE REVIEW

File: lbrynet/extras/daemon/Daemon.py [hunk 1]
```diff
@@ -2680,8 +2328,11 @@ def jsonrpc_channel_list(self, account_id=None, page=None, page_size=None):
         account = self.get_account_or_default(account_id)
         amount = self.get_dewies_or_error("amount", amount)
         result = await self.wallet_manager.support_claim(name, claim_id, amount, account)
-        await self.analytics_manager.send_claim_action('new_support')
-        return result
+        self.component_manager.analytics_manager.send_claim_action('new_support')
+        try:
+            return result
+        finally:
+            await self.component_manager.analytics_manager.send_claim_action('abandon')
```
Comment (eukreign) [inline lbrynet/extras/daemon/Daemon.py]: Why `abandon`?

File: setup.py [hunk 2]
```diff
@@ -25,7 +25,6 @@
     install_requires=[
         'aiohttp',
         'aioupnp',
-        'twisted[tls]==18.7.0',
```
Comment (eukreign) [inline setup.py]: :+1:

File: tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import logging
+logging.getLogger("asyncio").setLevel(logging.ERROR)
+# logging.getLogger("lbrynet").addHandler(logging.StreamHandler())
+# logging.getLogger("lbrynet").setLevel(logging.DEBUG)
```
Comment (eukreign) [inline tests/__init__.py]: what is going on here?

File: lbrynet/extras/daemon/Daemon.py [hunk 4]
```diff
@@ -2325,17 +1969,19 @@ def jsonrpc_stream_cost_estimate(self, uri, size=None):
...
+        try:
+            return {
+                "success": True,
+                "tx": tx,
+                "claim_id": txo.claim_id,
+                "claim_address": txo.get_address(self.ledger),
+                "output": txo
+            }
+        finally:
+           await self.component_manager.analytics_manager.send_new_channel()
```
[the remark this answers is not in the document]
Comment (jackrobison) [reply lbrynet/extras/daemon/Daemon.py]: Whoops, removed/removing these

File: lbrynet/extras/daemon/Daemon.py (hunk 1 above)
Comment (jackrobison) [reply lbrynet/extras/daemon/Daemon.py]: Will update the analytics calls today and make sure they're doing what they're supposed to

File: lbrynet/extras/daemon/Daemon.py [hunk 5]
```diff
@@ -61,9 +41,23 @@
...
+    from lbrynet.dht.node import Node
+
 log = logging.getLogger(__name__)
 
 
+#
+# async def download_single_blob(node: 'Node', blob_manager: 'BlobFileManager'):
+#     async with node.pe
+
+
```
[the remark this answers is not in the document]
Comment (jackrobison) [reply lbrynet/extras/daemon/Daemon.py]: Needs to be finished for `blob_get`, will do that today

File: lbrynet/blob_exchange/serialization.py [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+        return {
+            self.key: self.blob_data_payment_rate
+        }
+
+
+class BlobPriceResponse(BlobMessage):
+    key = 'blob_data_payment_rate'
+
+    def __init__(self, blob_data_payment_rate: str, **kwargs):
+        assert blob_data_payment_rate in ('RATE_ACCEPTED', 'RATE_TOO_LOW')
```
[the remark this answers is not in the document]
Comment (jackrobison) [reply lbrynet/blob_exchange/serialization.py]: Done

File: tests/__init__.py (hunk 3 above)
Comment (jackrobison) [reply tests/__init__.py]: Whoops, this should be with the rest of the logger setup

**Pull request merged**